File: backend/src/schemas/call_bell.py
from src.schemas.common import CamelModel


class CallBellEventResponse(CamelModel):
    id: str
    room_id: str
    room_number: str
    resident_id: str
    resident_name: str
    unit: str
    floor: int
    origin: str
    priority: str
    status: str
    vendor: str
    pressed_at: str
    responded_at: str | None
    closed_at: str | None
    response_time_seconds: int | None
    responded_by: str | None
    responded_by_name: str | None
    shift: str


class CallBellRespondRequest(CamelModel):
    user_id: str
    user_name: str


class CallBellStaffMetricsResponse(CamelModel):
    staff_id: str
    staff_name: str
    role: str
    total_calls: int
    avg_response_seconds: float
    median_response_seconds: float
    min_response_seconds: float
    max_response_seconds: float
    calls_under_60s: int
    calls_under_120s: int
    calls_over_180s: int


class CallBellFloorMetricsResponse(CamelModel):
    floor: int
    unit: str
    total_calls: int
    avg_response_seconds: float
    calls_by_priority: dict[str, int]
    calls_by_origin: dict[str, int]
    peak_hour: int


class CallBellShiftMetricsResponse(CamelModel):
    shift: str
    date: str
    unit: str
    total_calls: int
    avg_response_seconds: float
    responded_within_60s: int
    responded_within_120s: int
    slow_responses: int


class CallBellDailySummaryResponse(CamelModel):
    date: str
    total_calls: int
    avg_response_seconds: float

File: frontend/src/shared/types/callBell.ts
export type CallBellVendor = 'jeron' | 'rauland' | 'hill-rom';
export type CallBellStatus = 'active' | 'responded' | 'closed' | 'cancelled';
export type CallBellPriority = 'emergency' | 'urgent' | 'normal';
export type CallBellOrigin = 'bedside' | 'bathroom' | 'hallway' | 'pendant';

export interface CallBellEvent {
  id: string;
  roomId: string;
  roomNumber: string;
  residentId: string;
  residentName: string;
  unit: string;
  floor: number;
  origin: CallBellOrigin;
  priority: CallBellPriority;
  status: CallBellStatus;
  vendor: CallBellVendor;
  pressedAt: string;
  respondedAt: string | null;
  closedAt: string | null;
  responseTimeSeconds: number | null;
  respondedBy: string | null;
  respondedByName: string | null;
  shift: 'Day' | 'Evening' | 'Night';
}

export interface CallBellStaffMetrics {
  staffId: string;
  staffName: string;
  role: string;
  totalCalls: number;
  avgResponseSeconds: number;
  medianResponseSeconds: number;
  minResponseSeconds: number;
  maxResponseSeconds: number;
  callsUnder60s: number;
  callsUnder120s: number;
  callsOver180s: number;
}

export interface CallBellFloorMetrics {
  floor: number;
  unit: string;
  totalCalls: number;
  avgResponseSeconds: number;
  callsByPriority: Record<CallBellPriority, number>;
  callsByOrigin: Record<CallBellOrigin, number>;
  peakHour: number;
}

export interface CallBellShiftMetrics {
  shift: 'Day' | 'Evening' | 'Night';
  date: string;
  unit: string;
  totalCalls: number;
  avgResponseSeconds: number;
  respondedWithin60s: number;
  respondedWithin120s: number;
  slowResponses: number;
}

export interface CallBellDailySummary {
  date: string;
  totalCalls: number;
  avgResponseSeconds: number;
}
